### PR TITLE
Free redis connection on server stop

### PR DESCRIPTION
**Pull request by chuhaienko**

### CODE REVIEW

**Comment by marcuspoehls** [comment]
@chuhaienko Hi Yuri,

thank you for the PR, good idea to add the missing Redis connection close on server stop.

Can you please add tests for the new functionality?

Thank you!

**Comment by chuhaienko** [comment]
@marcuspoehls, Hi
Yes, I will try to add it today

**Comment by chuhaienko** [comment]
Done

**Comment by marcuspoehls** [comment]
Great! Thank you!
I'll review your code on Thursday. Will be back at work then

File: lib/index.js [hunk 1]
```diff
@@ -31,6 +32,13 @@ async function register (server, options) {
   server.ext('onPreResponse', (request, h) => {
     return limiter.addHeaders(request, h)
   })
+
+  /**
+   * Free Redis connection before exit
+   */
+  server.events.on('stop', async () => {
```
Comment (marcuspoehls) [inline lib/index.js]: Let's use the `onPostStop` extension point to close the connection:

```js
server.ext('onPostStop', () => {
  await limiter.closeRedis()
})
```

File: lib/index.js [hunk 2]
```diff
@@ -19,6 +19,7 @@ async function register (server, options) {
   server.decorate('request', 'rateLimit')
 
   const limiter = new RateLimiter(config)
+  server.expose('limiter', limiter)
```
Comment (marcuspoehls) [inline lib/index.js]: Are you exposing the limiter for testing?

**Comment by chuhaienko** [comment]
Hi, Marchus
The best is the enemy of the good)
Current change resolves the one problem i have touched. Redis connection still exists after server is down. It prevents mocha test to exit.

**Comment by marcuspoehls** [comment]
@chuhaienko I've tagged `2.6.0` and published on NPM. Enjoy!
Thank you for your contribution!

**Comment by chuhaienko** [comment]
@marcuspoehls , Super, Thank you)

**Pull request merged**